### ISSUE TITLE
Allow External TCP

### ISSUE THREAD

**Issue by mcsurfi** [boringproxy/boringproxy#25]
Hi

the option "Allow External TCP" does not work as expected. I've tried to forward a port to my local test pc to open rdp.

boringproxy "server" runs on a vps accessible via a public ip.
boringproxy "client" runs on a windows pc at home behind a nat.

I've created the following tunnel and checked the box "Allow External TCP".
--> host.mydomain.de:45871 -> PCatHome -> 127.0.0.1:3389

The tunnel are established between the server and the client, but the port on the server are only opend at the localhost IP:

netstat -na:
tcp        0      0 127.0.0.1:45871         0.0.0.0:*               LISTEN

How to config boringproxy to open the port on the public ip on the vps ?

Florian

**Comment by anderspitman** [comment]
Hi @mcsurfi. I just tried this on my personal instance. It appeared to work when doing Server-terminated TLS, but not Client-terminated. I'll need to look into this more. If you were using Client-terminated, can you try terminating at the server and see if that works?

**Comment by mcsurfi** [comment]
Hi 

I tried it with server and client terminated TLS. I'm unable to connect to my client with rdp....

![image](https://user-images.githubusercontent.com/10992142/103076818-d8ee2f80-45ce-11eb-9019-20474bd04f67.png)

Proto Recv-Q Send-Q Local Address           Foreign Address         State

tcp        0      0 127.0.0.1:37037         0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.1:33955         0.0.0.0:*               LISTEN

The port on the vps only listening on the localhost ip 127.0.0.1 but not on the public ip from the vps. In the config file the external tcp option is set "allow_external_tcp": true,". Maybe i missed something or there are other options i have to set that boringproxy are listen on the public ip. In the demo video you are talking about a special server option but i'm unable to find any information how i can set this option.

Florian

**Comment by anderspitman** [comment]
Oh of course! You need to set your sshd_config GatewayPorts to
clientspecified. Search this page for "GatewayPorts":
https://www.ssh.com/ssh/tunneling/example

For an explanation.

Let me know if that works.

**Comment by mcsurfi** [comment]
Hi

the "GatewayPorts" parameter was the missing hint. Maybe you could add this to your manual. I think other users will have the same problem.

Thank you for helping...

Florian

**Comment by anderspitman** [comment]
Yes that definitely needs to go in the docs! Thanks for the catch.

**Comment by luke2m** [comment]
https://github.com/boringproxy/boringproxy.io/pull/8 adds this to the website.

**Comment by luke2m** [comment]
@anderspitman this can be closed.
